### PR TITLE
Zephyr Quick Track Cleanup

### PR DESCRIPTION
**Pull request by D-Triveni**

### CODE REVIEW

File: eloop.c [hunk 1]
```diff
@@ -20,7 +20,11 @@
 #include <unistd.h>
 #include <errno.h>
 #include <signal.h>
+#ifdef CONFIG_ZEPHYR
+#include <zephyr/posix/sys/select.h>
```
Comment (krish2718) [inline eloop.c:24]: Please put this in a separate commit explaining why we need this?

File: eloop.c [hunk 2]
```diff
@@ -283,9 +287,9 @@ static void eloop_process_pending_signals(void)
 	eloop.signaled = 0;
 
 	if (eloop.pending_terminate) {
-#ifndef CONFIG_NATIVE_WINDOWS
+#if !defined(CONFIG_NATIVE_WINDOWS) && !defined(CONFIG_ZEPHYR)
```
Comment (krish2718) [inline eloop.c:290]: Please put this in a separate commit explaining why we need this?

File: zephyr/src/dut_main.c [hunk 3]
```diff
@@ -0,0 +1,249 @@
+/* Copyright (c) 2020 Wi-Fi Alliance                                                */
```
Comment (krish2718) [inline zephyr/src/dut_main.c]: Instead of copying OSS code, can we leverage those functions here and keep this file Zephyr specific (with nRF license)?

**Review by krish2718 — CHANGES_REQUESTED** [review]
I am still thinking about keeping non-zephyr code under a define (which never gets defined) vs. removing it? Esp. from maintenance PoV, if we fork it and customize heavily then keeping it up-to-date across upmerge's will get tricky.

Lets address other comments and circle back to this.

File: eloop.c [hunk 4]
```diff
@@ -287,9 +287,9 @@ static void eloop_process_pending_signals(void)
...
```
Comment (krish2718) [inline eloop.c:290]: If this commit is fixing another `nrf noup` please follow the procedure for that.

File: Makefile [hunk 5]
```diff
@@ -5,7 +5,7 @@ ROLE = dut
 # Package Version
 VERSION = "2.2.0.46"
 
-OBJS = main.o eloop.o indigo_api.o indigo_packet.o utils.o wpa_ctrl.o
+OBJS = main.o eloop.o indigo_api.o indigo_packet.o utils.o wpa_ctrl.o qt_client.o
```
Comment (krish2718) [inline Makefile:8]: missing sauce tag for this commit.

**Review by krish2718 — CHANGES_REQUESTED** [review]

File: eloop.c (hunk 4 above)
Comment (D-Triveni) [reply eloop.c:290]: @krish2718 
After adding dut initialization changes(dut_main.c), and I am seeing compilation issues. So, I have added this change. 
But I committed this change before and later  dut initialization changes. So i feel no fixup is required.
And also, let me know is it okay if commit order is like this.

Comment (krish2718) [reply eloop.c:290]: whichever commit we start seeing compilation failure, this fix should be part of it.

Comment (D-Triveni) [reply eloop.c:290]: InitIally I did that, u commented to split.

Comment (krish2718) [reply eloop.c:290]: Sorry, that looked like an unrelated fix, the commit log didn't explain, hence the confusion.

Comment (krish2718) [reply eloop.c:290]: so, it is fixing a `nrf noup`, so, you can keep this commit but follow the squashing policy.

**Review by krish2718 — APPROVED** [review]

File: zephyr/src/main.c [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+
+    /* Print the run-time information */
+    LOG_INF("QuickTrack control app running at: %d", get_service_port());
+    LOG_INF("Wireless Interface: %s", WIRELESS_INTERFACE_DEFAULT);
+
+    /* Register the callback */
+    register_apis();
+
+    /* Initiate the vendor's specific startup commands */
+    vendor_init();
```
Comment (krish2718) [inline zephyr/src/main.c]: Remove

File: zephyr/src/main.c [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+
+#include "vendor_specific.h"
+#include "eloop.h"
+#include "indigo_api.h"
+#include "utils.h"
+
+#include <zephyr/posix/pthread.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(wfa_qt, LOG_LEVEL_DBG);
```
Comment (krish2718) [inline zephyr/src/main.c]: Define a separate logger and use

File: zephyr/src/main.c [hunk 8]
```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <stdio.h>
+
+#include "vendor_specific.h"
```
Comment (krish2718) [inline zephyr/src/main.c]: no need

File: zephyr/src/main.c [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+                qt_main,
+                NULL,
+                NULL,
+                NULL,
+                0,
+                0,
+                0);
+
+static pthread_t main_thread;
+#define STACK_SIZE 4096
```
Comment (krish2718) [inline zephyr/src/main.c]: remove

File: zephyr/src/main.c [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+K_THREAD_STACK_DEFINE(main_thread_stack, STACK_SIZE);
```
Comment (krish2718) [inline zephyr/src/main.c]: Use WFA_QT_THREAD_STACK_SIZE, see my comment about two threads below.

File: zephyr/src/main.c [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+    pthread_attr_setschedpolicy(&ptAttr, SCHED_RR);
+    pthread_attr_getschedpolicy(&ptAttr, &ptPolicy);
+
+    ret = pthread_attr_setstack(&ptAttr, &main_thread_stack, 4096);
+    if (ret != 0) {
+        LOG_ERR("%s: pthread_attr_setstack failed: %d", __func__, ret);
+        return;
+    }
+
+    ret = pthread_create(&main_thread, &ptAttr, main_thread_handler, NULL);
```
Comment (krish2718) [inline zephyr/src/main.c]: Why do you need a second thread here? Stick to Zephyr native APIs for Zephyr code.

File: zephyr/src/vendor_specific_dut.c [hunk 12]
```diff
@@ -23,329 +23,68 @@
 #include "vendor_specific.h"
 #include "utils.h"
 
-#ifdef HOSTAPD_SUPPORT_MBSSID_WAR
```
Comment (krish2718) [inline zephyr/src/vendor_specific_dut.c:26]: Just remove this file. In a generic code it makes sense to call vendor specific, but this is Zephyr specific already, doesn't make sense, unless we use generic `indigo_*` files and implement `vendor_*` files for Zephyr.

File: zephyr/src/vendor_specific_dut.c [hunk 13]
```diff
@@ -354,150 +93,54 @@ int get_p2p_dev_if(char *if_name, size_t size) {
...
-                return __get_wps_setting(len, pipebuf, WPS_STA);
-            } else {
-                indigo_logger(LOG_LEVEL_INFO, "wsc settings STAUT: no data");
-            }
-        } else {
-            indigo_logger(LOG_LEVEL_ERROR, "STAUT: WPS Erorr. Failed to get settings.");
-            return NULL;
-        }
-    }
+    /*TODO:
```
Comment (krish2718) [inline zephyr/src/vendor_specific_dut.c]: Can you use a single line for all `TODO`?

**Review by krish2718 — CHANGES_REQUESTED** [review]

File: zephyr/src/vendor_specific_dut.c (hunk 12 above)
Comment (D-Triveni) [reply zephyr/src/vendor_specific_dut.c:26]: Based on discussion, keeping this file.

File: zephyr/src/main.c (hunk 7 above)
Comment (D-Triveni) [reply zephyr/src/main.c]: Cant'we use this? Any specific reason for new logger?

Comment (krish2718) [reply zephyr/src/main.c]: Yes, this is too generic and could enable other modules logs too. We need an option to enable logs only for this module.

File: zephyr/src/main.c (hunk 11 above)
Comment (D-Triveni) [reply zephyr/src/main.c]: Initially when QT isn't running from boot time, did this change to run app on a thread.
But now, we are running QT from boot time as a thread. So, no need for 2nd thread, will remove it and test.

File: zephyr/src/main.c (hunk 7 above)
Comment (D-Triveni) [reply zephyr/src/main.c]: @krish2718 Added, please review.

File: eloop.c (hunk 4 above)
Comment (krish2718) [reply eloop.c:290]: Your latest changes have overwritten mine, this and previous commit are missing the `fixup! [nrf noup]` line.

File: zephyr/CMakeLists.txt [hunk 14]
```diff
@@ -26,10 +26,12 @@ zephyr_include_directories_ifdef(CONFIG_WFA_QT_CONTROL_APP
...
 zephyr_library_sources_ifdef(CONFIG_WFA_QT_CONTROL_APP
 	# Zephyr's port of the Indigo API
+	${SOURCES_BASE}/zephyr/src/main.c
 	${SOURCES_BASE}/zephyr/src/indigo_api_callback_dut.c
 	${SOURCES_BASE}/zephyr/src/vendor_specific_dut.c
 	${SOURCES_BASE}/zephyr/src/utils.c
 	${SOURCES_BASE}/eloop.c
 	${SOURCES_BASE}/indigo_api.c
 	${SOURCES_BASE}/indigo_packet.c
+	${SOURCES_BASE}/qt_client.c
```
Comment (krish2718) [inline zephyr/CMakeLists.txt:36]: Commit title needs to cover all changes

File: zephyr/src/indigo_api_callback_dut.c [hunk 15]
```diff
@@ -1,4 +1,4 @@
-/* Copyright (c) 2020 Wi-Fi Alliance                                                */
+/* Copyright (c) 2023 Wi-Fi Alliance                                                */
```
Comment (krish2718) [inline zephyr/src/indigo_api_callback_dut.c]: we cannot change original year here as we don't own this copyright. We could add below, 

`/* Copyright (c) 2023 Nordic semiconductor ASA */`

**Review by krish2718 — CHANGES_REQUESTED** [review]

File: zephyr/src/indigo_api_callback_dut.c [hunk 16]
```diff
@@ -14,6 +14,8 @@
 /* OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS                          */
 /* SOFTWARE. */
 
+/* Copyright (c) 2023 Nordic semiconductor ASA */
```
Comment (krish2718) [inline zephyr/src/indigo_api_callback_dut.c]: you can add this next to the first copyright.

File: zephyr/Kconfig [hunk 17]
```diff
@@ -7,4 +7,5 @@
 config WFA_QT_CONTROL_APP
 	# Need full POSIX from libc, Zephyr's POSIX support is only partial
 	depends on !POSIX_API
+	select PTHREAD_IPC
```
Comment (krish2718) [inline zephyr/Kconfig:10]: This commit log also needs `fixup!`

File: zephyr/Kconfig [hunk 18]
```diff
@@ -8,4 +8,11 @@ config WFA_QT_CONTROL_APP
...
 	depends on !POSIX_API
 	select PTHREAD_IPC
+	select WFA_QT_LOG_LEVEL_DEFAULT
 	bool "WFA Quicktrack control app"
+
+config WFA_QT_THREAD_STACK_SIZE
+        int "Thread stack size"
+        default 4096
+        help
+          Thread stack size.
```
Comment (krish2718) [inline zephyr/Kconfig]: Remove this trivial and duplicate help or add a detailed help.

File: zephyr/Kconfig [hunk 19]
```diff
@@ -8,4 +8,11 @@ config WFA_QT_CONTROL_APP
 	# Need full POSIX from libc, Zephyr's POSIX support is only partial
 	depends on !POSIX_API
 	select PTHREAD_IPC
+	select WFA_QT_LOG_LEVEL_DEFAULT
 	bool "WFA Quicktrack control app"
+
+config WFA_QT_THREAD_STACK_SIZE
+        int "Thread stack size"
```
Comment (krish2718) [inline zephyr/Kconfig]: WFA QT Thread stack size

**Review by krish2718 — APPROVED** [review]
